tools/config/__init__.py

```diff
@@ -780,13 +780,29 @@ def _make_header_region(self, start, header_format, offset=None):
         return (start, region)
 
     @staticmethod
-    def _assign_new_offset(rom_start, start, new_offset, region_name):
+    def _assign_new_offset(rom_start, new_offset, region_name, regions):
         newstart = rom_start + integer(new_offset, 0)
-        if newstart < start:
-            raise ConfigException(
-                "Can not place %r region inside previous region" % region_name)
+
+        for s, e in regions:
+            if newstart > s and newstart < e:
+                raise ConfigException(
+                    "Can not place %r region inside previous region" % region_name)
         return newstart
 
+    @staticmethod
+    def _get_end_address(region_list, start_address, rom_end):
+        """Given a start address and set of regions, compute
+        the end address. The end address is either
+        rom_end or beginning of next section, whichever
+        is smaller
```

Comment:
We should document that we expect the `region_list` to be sorted based on start address. OR you can always sort the regions in this function to ensure that.

Reply:
Good point. Uploaded a patch set with suggested changes. Please re-review